source/server/admin/logs_handler.cc

```diff
@@ -56,56 +60,96 @@ bool LogsHandler::changeLogLevel(const Http::Utility::QueryParams& params) {
     return false;
   }
 
-  std::string name = params.begin()->first;
-  std::string level = params.begin()->second;
+  const auto it = params.begin();
+  absl::string_view key = it->first;
+  absl::string_view value = it->secondn;
+
+  if (key == "paths") {
+    // Bulk change log level by name:level pairs
+    std::vector<absl::string_view> pairs = absl::StrSplit(value, ',', absl::SkipWhitespace());
+    bool ret = true;
+    for (const auto& name_value : pairs) {
+      std::pair<absl::string_view, absl::string_view> name_value_pair =
+          absl::StrSplit(name_value, absl::MaxSplits(':', 1), absl::SkipWhitespace());
+
+      auto name = name_value_pair.first;
+      auto value = name_value_pair.second;
+
+      if (name.empty() || value.empty()) {
+        continue;
+      }
+
+      ret = changeLogLevelByName(name, value) && ret;
```

Comment:
nit: this is n^2 -- suggest making a `absl::flat_hash_map<absl::string_view,absl::string_view>` here, and passing that map once to `changeLogLevelsByName(map<...>& changes)` so it can iterate once through the loggers.

You might argue there are so few loggers that this n^2 loop doesn't matter and I'd be OK with that, but call that out in a comment as a decision you've made. But it might be just as easy and clean to make the map as to explain why it's probably OK not to :)